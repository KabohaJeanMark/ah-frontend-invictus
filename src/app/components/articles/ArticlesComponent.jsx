import React from 'react';
import { Link } from "react-router-dom";
import '../../../assets/styles/articles.scss';

const ArticlesComponent = (props) => {
  const articlesItems = props.state.articles.articles.map(article => (
    <Link key={article.id} className="card-article-link" to={`/articles/${article.id}`}>
      <div className="card-article" style={{ width: 21.85 + 'rem' }}>
        <img className="card-image" alt="" src="https://picsum.photos/600/350" />
        <div className="card-body">
          <div><p className="card-publish-date">Published On: {article.created_at.substring(0, 10)}</p></div>
          <h4 className="card-title">{article.title}</h4><p className="card-text">{article.description}</p>
          <div className="card-body-footer">
            <p className="card-footer-text">@{article.author}</p>
            <p className="card-footer-text">{article.read_time}</p></div>
          <div className="card-body-likes-dislikes">
            <p className="card-body-likes-text">Likes: {' '}{article.likes_count}</p>
            <p className="card-body-dislikes-text">Dislikes: {' '}{article.dislikes_count}</p></div>
          <div className="all-tags"><hr />
            <ul>{article.tagList.map(tag => <li key={tag} className="tag">{tag}</li>)}</ul></div>
        </div></div>
    </Link>));
  return (
    <div>
      <div className="container">
        <h2 className="articles-heading">Recent Articles</h2><hr />
        <div className="row">{articlesItems}</div>
      </div>
    </div>
  );
};

export default ArticlesComponent;
