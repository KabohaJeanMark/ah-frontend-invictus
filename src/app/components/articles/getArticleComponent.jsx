import React from 'react';
import Navbar from '../../components/Navbar';
import CommentContainer from '../../containers/CommentContainer';

const ArticleComponent = (props) => {
  const { articleId, article: { title, body, author, description, tagList = [] }, handleLike, handleDislike } = props;
  return (
    <div>
      <Navbar />
      <div className="container">
        <div className="article-main">
          <h1 className="article-title">{title}</h1>
          <div className="article-desc">{description}</div>
          <div>{author}</div>
          <img className="card-image" alt="" src="https://picsum.photos/600/300" />
          <div className="article-body">{body}</div>
          <div className="like-dislike">
            <i className="fa fa-thumbs-up" onClick={handleLike} role="presentation"
              style={{ color: 'blue', marginRight: '10px' }} />
            <i
              className="fa fa-thumbs-down" onClick={handleDislike} role="presentation" style={{ color: 'red' }}
            /></div>
          <div className="all-tags"><hr />
            <ul>{tagList.map(tag => <li key={tag} className="tag">{tag}</li>)}</ul>
            <div>
              <CommentContainer articleId={articleId} />
            </div>
          </div>
        </div>
      </div>
    </div >
  );
};
export default ArticleComponent;
