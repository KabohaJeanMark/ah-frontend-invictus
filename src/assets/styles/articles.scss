@import "./var";

.articles-heading {
  font-weight: bold;
  margin-top: 30px;
}

.card-article-link:hover {
  text-decoration: none;
}

.card-article {
  background-color: $article-card-bacground;
  color: $text-color;
  margin: 15px;
}

.card-image {
  width: 100%;
}

.card-title {
  font-weight: bold;
}

.card-publish-date {
  color: $text-color-secondary;
  font-size: 13px;
}

.card-body-footer {
  display: flex;
}

.card-footer-text {
  font-size: 13px;
  margin-right: 5px;
}

.article-main {
  padding-left: 150px;
  padding-right: 150px;
  padding-top: 80px;
}

.article-title {
  font-weight: bold;
}

.article-body {
  margin-bottom: 30px;
  margin-top: 30px;
}

.article-desc {
  color: $text-color-secondary-second;
  font-size: 26px;
  margin-bottom: 20px;
  margin-top: 20px;
}
