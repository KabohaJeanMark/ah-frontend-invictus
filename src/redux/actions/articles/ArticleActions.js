import { articleTypes } from '../../actions/types';
import axios from 'axios';
import { fetchArticleUrl } from '../../../app/urls';

let authorizationHeader = localStorage.getItem('user_token');

const fetchArticle = (id) => dispatch => {
  return axios.get(`${fetchArticleUrl}${id}/`, {
    headers: { Authorization: `Bearer ${authorizationHeader}` }
  })

    .then(response => {
      console.log(response.data.article);
      dispatch({
        type: articleTypes.FETCH_ARTICLE,
        payload: response.data.article
      });
    });
};

export default fetchArticle;
