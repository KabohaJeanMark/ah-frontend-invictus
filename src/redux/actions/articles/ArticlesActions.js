import { articlesTypes } from '../types';
import axios from 'axios';
import { fetchArticlesUrl } from '../../../app/urls';

let authorizationHeader = localStorage.getItem('user_token');

const fetchArticles = () => dispatch => {
  return axios.get(fetchArticlesUrl, {
    headers: { Authorization: `Bearer ${authorizationHeader}` }
  })
    .then(response => {
      dispatch({
        type: articlesTypes.FETCH_ARTICLES,
        payload: response.data.article.results
      });
    });
};

export default fetchArticles;
