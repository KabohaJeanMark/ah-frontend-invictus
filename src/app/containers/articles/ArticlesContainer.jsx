import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ArticlesComponent from '../../../app/components/articles/ArticlesComponent';
import fetchArticles from '../../../redux/actions/articles/ArticlesActions';

export class Articles extends Component {
  componentDidMount () {
    this.props.fetchArticles();
  }

  render () {
    return (
      <div>
        <ArticlesComponent state={this.props} />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  articles: state.articles
});

Articles.propTypes = {
  fetchArticles: PropTypes.func
};

export default connect(mapStateToProps, { fetchArticles })(Articles);
