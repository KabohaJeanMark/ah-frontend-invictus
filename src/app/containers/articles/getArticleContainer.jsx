import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import GetArticleComponent from '../../../app/components/articles/getArticleComponent';
import fetchArticle from '../../../redux/actions/articles/ArticleActions';
import { likeArticle, dislikeArticle } from '../../../redux/actions/ArticleActions';

export class Article extends Component {
  componentDidMount () {
    const articleId = this.props.match.params.id;
    this.props.fetchArticle(articleId);
  }

  handleLike = () => {
    const articleId = this.props.match.params.id;
    this.props.likeArticle(articleId, this.props);
  };

  handleDislike = () => {
    const articleId = this.props.match.params.id;
    this.props.dislikeArticle(articleId, this.props);
  };

  render () {
    return (
      <div>
        <GetArticleComponent {...this.props.article}
          handleLike={this.handleLike}
          handleDislike={this.handleDislike}
        />
      </div>
    );
  }
}


export const mapStateToProps = state => ({
  article: state.article
});

Article.propTypes = {
  fetchArticle: PropTypes.func,
  handleLike: PropTypes.func,
  handleDislike: PropTypes.func,
  likeArticle: PropTypes.func,
  dislikeArticle: PropTypes.func
};

export default connect(mapStateToProps, { fetchArticle, likeArticle, dislikeArticle })(Article);
