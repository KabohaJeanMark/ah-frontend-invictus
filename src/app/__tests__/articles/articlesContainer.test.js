import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Articles, mapStateToProps } from '../../containers/articles/ArticlesContainer';

const storeItems = {};
const mockStore = configureStore();
const store = mockStore(storeItems);

const props = {
  ArticlesReducer: {
    articles: []
  },
  fetchArticles: jest.fn(),
};

describe('fetch articles', () => {
  const mockFetch = jest.fn();
  afterEach(() => {
    mockFetch.mock.calls = []
  });

  it('should render correctly', () => {
    const wrapper = shallow(<Articles store={store} {...props} />); 
    expect(toJson(wrapper)).toMatchSnapshot();  
  });
  
  it('testing mapStateToProps', () => {
    const initialState = {articles: []};
    expect(mapStateToProps(initialState)).toEqual({articles: []});
  });
});
