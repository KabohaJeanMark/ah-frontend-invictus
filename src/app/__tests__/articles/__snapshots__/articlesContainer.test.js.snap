// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch articles should render correctly 1`] = `
<div>
  <ArticlesComponent
    state={
      Object {
        "ArticlesReducer": Object {
          "articles": Array [],
        },
        "fetchArticles": [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      }
    }
  />
</div>
`;
